x-pack/plugins/cloud_security_posture/server/lib/fleet_util.ts

```diff
@@ -33,22 +33,24 @@ const getPackageNameQuery = (packageName: string, benchmarkFilter?: string): str
   return kquery;
 };
 
-export const addRunningAgentToAgentPolicy = async (
+export type AgentStatusByAgentPolicyMap = Record<string, GetAgentStatusResponse['results']>;
+
+export const getAgentStatusesByAgentPolicies = async (
   agentService: AgentService,
-  agentPolicies: AgentPolicy[]
-): Promise<GetAgentPoliciesResponseItem[]> => {
-  if (!agentPolicies.length) return [];
+  agentPolicies: AgentPolicy[] | undefined
+): Promise<AgentStatusByAgentPolicyMap> => {
+  if (!agentPolicies?.length) return {};
+
+  const internalAgentService = agentService.asInternalUser;
+  const result: AgentStatusByAgentPolicyMap = {};
+
+  for (const agentPolicy of agentPolicies) {
+    result[agentPolicy.id] = await internalAgentService.getAgentStatusForAgentPolicy(
+      agentPolicy.id
+    );
+  }
```

Comment:
prefer using `agentPolicies.reduce`, or alternatively use a `forEach` loop instead of `for`.